internal/operators/pm.go

```diff
@@ -9,30 +9,57 @@ import (
 	"strings"
 
 	"github.com/corazawaf/coraza/v3/rules"
-
-	"github.com/corazawaf/coraza-proxy-wasm/internal/ahocorasick"
+	ahocorasick "github.com/wasilibs/go-aho-corasick"
 )
 
 type pm struct {
-	m ahocorasick.Matcher
+	matcher ahocorasick.AhoCorasick
 }
 
 var _ rules.Operator = (*pm)(nil)
 
 func newPM(options rules.OperatorOptions) (rules.Operator, error) {
-	return &pm{m: ahocorasick.NewMatcher(strings.Split(options.Arguments, " "))}, nil
+	data := options.Arguments
+
+	data = strings.ToLower(data)
+	dict := strings.Split(data, " ")
+	builder := ahocorasick.NewAhoCorasickBuilder(ahocorasick.Opts{
+		AsciiCaseInsensitive: true,
+		MatchOnlyWholeWords:  false,
+		MatchKind:            ahocorasick.LeftMostLongestMatch,
+		DFA:                  true,
+	})
+
+	// TODO this operator is supposed to support snort data syntax: "@pm A|42|C|44|F"
+	return &pm{matcher: builder.Build(dict)}, nil
 }
 
 func (o *pm) Evaluate(tx rules.TransactionState, value string) bool {
-	return pmEvaluate(o.m, tx, value)
+	return pmEvaluate(o.matcher, tx, value)
 }
 
-func pmEvaluate(m ahocorasick.Matcher, tx rules.TransactionState, value string) bool {
-	matches := m.Matches(value, 8)
-	if tx.Capturing() {
-		for i, c := range matches {
-			tx.CaptureField(i, c)
+func pmEvaluate(matcher ahocorasick.AhoCorasick, tx rules.TransactionState, value string) bool {
```

Comment:
I wonder if it is a good timing to start with unit tests for this.

Reply:
This code is copied exactly from Coraza with just import change - let's add unit tests there first